backend/pkg/api/data_access/notifications.go

```diff
@@ -34,8 +42,169 @@ type NotificationsRepository interface {
+			goqu.Ex{"uvd.user_id": userId},
+			goqu.L("uvd.network = ANY(?)", pq.Array(chainIds)),
+		).
+		GroupBy(
+			goqu.I("uvdnh.epoch"),
+			goqu.I("uvd.network"),
+			goqu.I("uvd.id"),
+			goqu.I("uvdg.id"),
+			goqu.I("uvdg.name"),
+		)
+
+	// TODO account dashboards
+	/*adbQuery := goqu.Dialect("postgres").
+		From(goqu.T("adb_notifications_history").As("anh")).
+		Select(
+			goqu.L("true").As("is_account_dashboard"),
+			goqu.I("anh.network").As("chain_id"),
+			goqu.I("anh.epoch"),
+			goqu.I("uad.id").As("dashboard_id"),
+			goqu.I("uad.name").As("dashboard_name"),
+			goqu.I("uadg.id").As("group_id"),
+			goqu.I("uadg.name").As("group_name"),
+			goqu.SUM("anh.event_count").As("entity_count"),
+			goqu.L("ARRAY_AGG(DISTINCT event_type)").As("event_types"),
+		).
+		InnerJoin(goqu.T("users_acc_dashboards").As("uad"), goqu.On(
+			goqu.Ex{"uad.id": goqu.I("anh.dashboard_id"),
+			})).
+		InnerJoin(goqu.T("users_acc_dashboards_groups").As("uadg"), goqu.On(
+			goqu.Ex{"uadg.id": goqu.I("anh.group_id"),
+			goqu.Ex{"uadg.dashboard_id": goqu.I("uad.id")},
+			})).
+		Where(
+			goqu.Ex{"uad.user_id": userId},
+			goqu.L("anh.network = ANY(?)", pq.Array(chainIds)),
+		).
+		GroupBy(
+			goqu.I("anh.epoch"),
+			goqu.I("anh.network"),
+			goqu.I("uad.id"),
+			goqu.I("uadg.id"),
+			goqu.I("uadg.name"),
+		)
+
+	unionQuery := vdbQuery.Union(adbQuery)*/
+	unionQuery := goqu.From(vdbQuery)
+
+	// sorting
+	defaultColumns := []t.SortColumn{
+		{Column: enums.NotificationDashboardTimestamp.ToString(), Desc: true, Offset: currentCursor.Epoch},
+		{Column: enums.NotificationDashboardDashboardName.ToString(), Desc: false, Offset: currentCursor.DashboardName},
+		{Column: enums.NotificationDashboardGroupName.ToString(), Desc: false, Offset: currentCursor.GroupName},
+		{Column: enums.NotificationDashboardChainId.ToString(), Desc: true, Offset: currentCursor.ChainId},
+	}
+	var offset any
+	if currentCursor.IsValid() {
+		switch colSort.Column {
+		case enums.NotificationDashboardTimestamp:
+			offset = currentCursor.Epoch
+		case enums.NotificationDashboardDashboardName:
+			offset = currentCursor.DashboardName
+		case enums.NotificationDashboardGroupName:
+			offset = currentCursor.GroupName
+		case enums.NotificationDashboardChainId:
+			offset = currentCursor.ChainId
+		}
+	}
+	order, directions := applySortAndPagination(defaultColumns, t.SortColumn{Column: colSort.Column.ToString(), Desc: colSort.Desc, Offset: offset}, currentCursor.GenericCursor)
+	unionQuery = unionQuery.Order(order...)
+	if directions != nil {
+		unionQuery = unionQuery.Where(directions)
+	}
+
+	// search
+	searchName := regexp.MustCompile(`^[a-zA-Z0-9_\-.\ ]+$`).MatchString(search)
+	if searchName {
+		searchLower := strings.ToLower(strings.Replace(search, "_", "\\_", -1)) + "%"
+		unionQuery = unionQuery.Where(exp.NewExpressionList(
+			exp.OrType,
+			goqu.L("LOWER(?)", goqu.I("dashboard_name")).Like(searchLower),
+			goqu.L("LOWER(?)", goqu.I("group_name")).Like(searchLower),
```

Comment:
**suggestion:** You could use `ILIKE` so you don't need to lower the strings.

Reply:
According to my tests a while back `LOWER` seems to be slightly faster, you can also find similar results online (e.g. https://stackoverflow.com/a/20336666). Maybe a preference thing but I'd keep it for that reason